packages/protocol/contracts/L1/TaikoData.sol

```diff
@@ -64,6 +64,8 @@ library TaikoData {
         bytes32 parentMetaHash;
         HookCall[] hookCalls; // DEPRECATED, value ignored.
         bytes signature;
+        uint32 l1StateBlockNumber;
```

Comment:
in order to merge this PR to main in the future, I think we should rename this struct to BlockParams2 and keep BlockParams as-is so we can support hardfork logics in the protocol.